onnxruntime/core/session/inference_session.h

```diff
@@ -434,5 +434,13 @@ class InferenceSession {
 #ifdef ENABLE_LANGUAGE_INTEROP_OPS
   InterOpDomains interop_domains_;
 #endif
+
+  // used to support platform telemetry 
+  static std::atomic<uint32_t> global_session_id_;  // a monotonically increasing session id
+  uint32_t session_id_;                             // the current session's id
+  uint32_t total_runs_since_last_;                  // the total number of Run() calls since the last report
+  long long total_run_duration_since_last_;         // the total duration (us) of Run() calls since the last report
+  TimePoint time_sent_last_;                        // the TimePoint of the last report
+  const long long kDurationBetweenSending = 1000* 1000 * 60 * 10;  // duration in (us).  send a report every 10 mins
```

Comment:
Should this be configurable?

Reply:
we could add a feature to make it configurable, but what we have found is that we never really change it.   we've gone a couple of years on some of our events without changing the duration sent.  we don't really want app developers to experiment too much with this number, as it becomes complex as to what guidance we would give them on how to experiment.   it's not a runtime perf thing.   there is some factor of network traffic and queue management, which is why we usually pick numbers like "10 minutes" .  something not too fast as to worry about network congestion or queues, and fast enough that we get regular heartbeat signal.

---
In reply to: [335301525](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335301525) [](ancestors = 335301525)